modules/placement-driver/src/main/java/org/apache/ignite/internal/placementdriver/LeaseUpdater.java

```diff
@@ -368,15 +369,14 @@ private void updateLeaseBatchInternal() {
                         writeNewLease(grpId, lease, candidate, renewedLeases, toBeNegotiated);
 
                         continue;
+                    } else {
+                        agreement.checkValid(grpId, topologyTracker.currentTopologySnapshot(), assignments);
```

Comment:
I would put it up just after getting an agreement with the additional conditions.
It makes logic simpler.

Reply:
we don't need to check the validity if the agreement is ready (accepted also means ready), that's why I put it to `else` block